@here/harp-map-controls/lib/CameraAnimationBuilder.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates } from "@here/harp-geoutils";
+import { MapView, MapViewUtils } from "@here/harp-mapview";
+import THREE = require("three");
+
+import { CameraKeyTrackAnimationOptions, ControlPoint } from "./CameraKeyTrackAnimation";
+import { CameraPose } from "./Utils";
+
+/**
+ * A Utility class for the creation of Camera Animations
+ */
+export class CameraAnimationBuilder {
+    /**
+     * Generates a ControlPoint from the Current View for the usage in a [[CameraKeyTrackAnimation]]
+     *
+     * @param mapView The MapView
+     */
+    static getPoseFromView(mapView: MapView): CameraPose {
+        const sphericalCoordinates = MapViewUtils.extractSphericalCoordinatesFromLocation(
+            mapView,
+            mapView.camera,
+            mapView.target
+        );
+        const cameraPose: CameraPose = new CameraPose({
+            geoCoordinate: mapView.target,
+            azimuth: THREE.MathUtils.radToDeg(Math.PI + sphericalCoordinates.azimuth),
+            altitude: THREE.MathUtils.radToDeg(sphericalCoordinates.tilt),
+            distance: mapView.targetDistance
+        });
+
+        return cameraPose;
+    }
+
+    /**
+     * Sets the view to the defiend [[CameraPose]]
+     */
+    static setViewFromPose(mapView: MapView, pose: CameraPose) {
+        mapView.lookAt({
+            target: pose.geoCoordinate,
+            distance: pose.distance,
+            tilt: pose.direction.altitude,
+            heading: pose.direction.azimuth
+        });
+    }
+
+    /**
+     * Appends a [[ControlPoint]] to [[CameraKeyTrackAnimationOptions]]
+     *
+     * @param options
+     * @param point
+     */
+    static appendControlPoint(options: CameraKeyTrackAnimationOptions, point: ControlPoint) {
+        const timestamp = options.controlPoints[options.controlPoints.length - 1].timestamp + 10;
```

Comment:
what does the +10 mean? Is it ms or seconds? Should it be configurable? Or is the default satifisfactory?

Reply:
added a descriptive variable instead